### PR TITLE
[feat] support attribute (name, value) filtering

### PR DESCRIPTION
**Pull request by ephetic**
See https://github.com/antonmedv/finder/issues/16

### CODE REVIEW

**Comment by ephetic** [comment]
@antonmedv The snapshot tests are failing on master and thus this branch as well, but I don't know how to fix them.

**Comment by ephetic** [comment]
This test update is just to include the block generated for `config:attr`.  The problem causing the tests the fail on master (and thus this branch) remains.

```
$ node -v
v8.9.4
$ npm -v
5.6.0
```

File: src/index.ts [hunk 1]
```diff
@@ -197,7 +197,7 @@ function attr(input: Element): Node[] {
 
   return attrs.map((attr): Node => ({
     name: '[' + cssesc(attr.name, {isIdentifier: true}) + '="' + cssesc(attr.value) + '"]',
-    penalty: 1
+    penalty: 0.5
```
Comment (antonmedv) [inline src/index.ts]: Why 0.5?

Comment (ephetic) [reply src/index.ts]: So that it gets prefered to classes but not ids.

Comment (antonmedv) [reply src/index.ts]: Okay, I think attributes may be more important when classes. Maybe create another PR with config option for it?

File: src/index.ts [hunk 2]
```diff
@@ -19,6 +19,7 @@ export type Options = {
   idName: (name: string) => boolean
   className: (name: string) => boolean
   tagName: (name: string) => boolean
+  attr: (name: string, value: any) => boolean
```
Comment (antonmedv) [inline src/index.ts]: value is `string` too.

**Pull request merged**